### PR TITLE
Pin libprotobuf<5

### PR DESCRIPTION
**Pull request by wrieg123**
Checklist
* [x] Used a [personal fork of the feedstock to propose changes](https://conda-forge.org/docs/maintainer/updating_pkgs.html#forking-and-pull-requests)
* [x] Bumped the build number (if the version is unchanged)
* [ ] Reset the build number to `0` (if the version changed)
* [ ] [Re-rendered]( https://conda-forge.org/docs/maintainer/updating_pkgs.html#rerendering-feedstocks ) with the latest `conda-smithy` (Use the phrase <code>@<space/>conda-forge-admin, please rerender</code> in a comment in this PR for automated rerendering)
* [ ] Ensured the license file is being packaged.

Related to [build issues](https://github.com/Point72/csp/issues/367#issuecomment-2408013630) with libprotobuf 5+, this pins the version back.

### CODE REVIEW

**Comment by wrieg123** [comment]
Specifically related to the code changes noted [here](https://github.com/conda-forge/csp-feedstock/pull/38#issuecomment-2385950081).

**Comment by timkpaine** [comment]
The solver will just take build number `0` to work around this constraint, it would really need to be added to the [metadata repo](https://github.com/conda-forge/conda-forge-repodata-patches-feedstock)

**Comment by wrieg123** [comment]
> The solver will just take build number `0` to work around this constraint, it would really need to be added to the [metadata repo](https://github.com/conda-forge/conda-forge-repodata-patches-feedstock)

I added [a PR](https://github.com/conda-forge/conda-forge-repodata-patches-feedstock/pull/881) to the repodata-patches-feedstock, if you wouldn't mind taking a look.

**Review by xhochy — CHANGES_REQUESTED** [review]
You should be pinning to an explicit version in host, otherwise you won't get reproducible/compatible builds.

**Comment by xhochy** [comment]
A rerender should also show whether this changes something. If you haven't applied a migration recently, you probably don't need this fix.

**Comment by timkpaine** [comment]
We're going to add support upstream for either major version

File: recipe/meta.yaml [hunk 1]
```diff
@@ -64,7 +64,7 @@ requirements:
     - libarrow =16
     - libboost-headers
     - libparquet =16
-    - libprotobuf
+    - libprotobuf =4.25.3
```
Comment (xhochy) [inline recipe/meta.yaml:67]: This is the currently pinned version: https://github.com/conda-forge/csp-feedstock/blob/834f80f49e544c611079730df0e6a37449344c2e/.ci_support/linux_64_numpy1.22python3.10.____cpython.yaml#L22 Don't add a pin here as this stop future migrations. Just don't merge migration PRs for not.

**Review by xhochy — CHANGES_REQUESTED** [review]

File: recipe/meta.yaml (hunk 1 above)
Comment (wrieg123) [reply recipe/meta.yaml:67]: I am missing some context here. If I don't pin the libprotobuf version here to either `<5` or specific version  `4.*.*` the builds all fail.

Comment (xhochy) [reply recipe/meta.yaml:67]: No, they shouldn't fail as this will take the global pin which is at the 4.25.t.

Comment (wrieg123) [reply recipe/meta.yaml:67]: Right, if I remove this global pin it won't take the other versions.

**Pull request closed without merging**